### PR TITLE
Use Chatbot for stench levels part 2

### PR DESCRIPTION
**Pull request by careyourcurroundings**
Alium and I worked on it simultaneously without knowing. 
see: https://github.com/tigerinanenet/cognac/pull/22
Definitely compiles, and should be good code, but hard to test without breaking TOS.
Chatbot code def copies to output folder, so if Alium's code is better, you can copy that part to it, and close this PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,6 +20,13 @@ git checkout https://github.com/tigerinanenet/cognac.git release
 
 Properties you can set in KoLMafia to customize the way `cognac` runs.
 
+### cognac_chatbot
+
+Required - Set the chatbot in kolmafia CLI:
+```
+set chatbotScript=scripts\\cognac\\stenchtracker.js
```
Comment (Rinn) [inline README.md:27]: why don't you just set this in code when cognac starts up?

File: src/lib/gossip.ts [hunk 2]
```diff
@@ -153,7 +153,7 @@ export class Gossip {
 
   readyToDive(): boolean {
     return (
-      this.stench >= BASE_STENCH_REQUIRED + this.players.length + this.requestingCompost.length
+      parseInt(getProperty("_heapStench")) >= BASE_STENCH_REQUIRED + Math.ceil(this.players.length * 1.1) + this.requestingCompost.length
```
Comment (Alium58) [inline src/lib/gossip.ts]: Was this to add margin?

File: src/scripts/stenchtracker.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+var kolmafia = require('kolmafia');
+
+module.exports.main = function main(sender, message, channel)
+{
+	if (channel == "/hobopolis" && sender == "Dungeon")
+	{
+		if (message.endsWith("moved some trash out of the Purple Light District."))
+		{
+			let stench = kolmafia.getProperty("_heapStench");
+			kolmafia.setProperty("_heapStench", stench == "" ? 0 : parseInt(stench) + 1);
```
Comment (Alium58) [inline src/scripts/stenchtracker.js]: I changed this to default to 1 instead of 0. Agree that is correct?

**Comment by tigerinanenet** [comment]
Thank you for your contributions, you two!

I haven't really reviewed the code yet, but there are a couple of ideas I wanted to bring up

- Can we make sure that a new joiner doesn't spin endlessly in the heap? We can accomplish this by waiting for a new round to drop whenever we join.
- To drop the mutex, we can't be using the whiteboard anymore. We would also need to handle communicating our need to compost in chat. We can have this chat functionality be silent for the time being, that is leaving `gossip.ts` as is, that way we can merge this chunk in once its ready to go.

LMK what you think, and LMK when you are ready for a proper review!

**Comment by Alium58** [comment]
@tigerinanenet believe mine is ready for the real review, #22. 

Added support for initial stench level to be set to whiteboard so new folks can join mid cycle
Agree this change along won't allow us to remove the mutex

**Comment by careyourcurroundings** [comment]
Probably close this and move convo to alium's pull.

**Comment by Alium58** [comment]
can be closed to #22

**Pull request closed without merging**